File: uc/computerScience2/spring2014/liveDemos/week03-pizza/pizza-05/pizza.cpp
#include "pizza.h"

Pizza::Pizza() {
  // nothing to initialize in the constructor
}
 
void Pizza::addTopping(Topping topping) {
  this->toppings.push_back(topping);
}

vector<Topping> Pizza::getToppings() {
  return toppings;
}

int Pizza::getToppingsCount() {
  return toppings.size();
}

double Pizza::getPrice() {
  double price = 8.99; // base price

  // for (int i = 0; i < toppings.size(); i++) {
  //  if (toppings[i].isPremium()) {

  for (vector<Topping>::iterator iter = toppings.begin(); iter != toppings.end(); iter++) {
    if (iter->isPremium()) {
      price += 1.25;
    } else {
      price += 0.75;
    }
  }
  return price;
}

File: uc/computerScience2/fall2013/liveDemos/java/pizza/Pizza.java

import java.util.List;
import java.util.ArrayList;

public class Pizza {
  private List<Topping> toppings = new ArrayList<Topping>();

  public Pizza() {
  }

  public void addTopping(Topping topping) {
    toppings.add(topping);
  }

  public List<Topping> getToppings() {
    return toppings;
  }

  public int getToppingsCount() {
    return toppings.size();
  }
  
  public double getPrice() {
    double price = 8.99; // base price
    for (Topping topping : toppings) {
      if (topping.isPremium()) {
        price += 1.25;
      } else {
        price += 0.75;
      }
    }
    return price;
  }
}